engines/wintermute/base/base_persistence_manager.cpp

```diff
@@ -821,6 +821,64 @@ bool BasePersistenceManager::transferVector2(const char *name, Vector2 *val) {
 	}
 }
 
+#ifdef ENABLE_WME3D
+//////////////////////////////////////////////////////////////////////////
+// Vector3
+bool BasePersistenceManager::transferVector3d(const char* name, Math::Vector3d* val) {
+	if (_saving) {
+		putFloat(val->x());
+		putFloat(val->y());
+		putFloat(val->z());
+
+		if (_saveStream->err()) {
+			return STATUS_FAILED;
+		}
+
+		return STATUS_OK;
+	} else {
+		val->x() = getFloat();
+		val->y() = getFloat();
+		val->z() = getFloat();
+
+		if (_loadStream->err()) {
+			return STATUS_FAILED;
+		}
+
+		return STATUS_OK;
+	}
+}
+
+//////////////////////////////////////////////////////////////////////////
+// Matrix4
+bool BasePersistenceManager::transferMatrix4(const char* name, Math::Matrix4* val) {
+	if (_saving) {
+		for (int i = 0; i < 4; ++i) {
```

Comment:
I have a preference for using loop-variables that make it clearer what's row and what's column. For instance "r" and "c".